includes/writer.js

```diff
@@ -54,26 +53,16 @@ module.exports = function(couchDbUrl, bufferSize, parallelism) {
     });
 
     response.on('response', function(resp) {
-      var e = null;
-      switch (resp.statusCode) {
-        // TODO move this to a common file where it can be re-used:
-        // deal with all expected error codes and distinguish between
-        // permanent and transient errors
-        case 401:
-          e = new error.BackupError('Unauthorized', `Database ${couchDbUrl} does not have correct permissions. Check that you have the correct permissions before restoring.`);
-          break;
-        case 403:
-          e = new error.BackupError('Forbidden', `Incorrect credentials for database ${couchDbUrl}. Check that you have the correct username and password or API key before restoring.`);
-          break;
-      }
-      if (e != null) {
-        response.abort();
-        // only emit the first error as there are multiple threads
-        if (!didError) {
-          didError = true;
-          writer.emit('error', e);
+      request.isOk(resp, function(err) {
```

Comment:
I've lost the plot a little since the changes while I was out, but does this actually retry like backup does? I had a feeling it didn't and if it doesn't shouldn't this be fatal, on the basis that some docs will now be missing from the restored DB?